src/json-rpc-handlers/dwn/process-message.ts

```diff
@@ -62,6 +62,8 @@ export const handleDwnProcessMessage: JsonRpcHandler = async (
       return { jsonRpcResponse };
     }
 
+    // if this is a subscription request, we first check if the connection has a subscription with this Id
+    // we do this ahead of time to prevent opening a subscription on the dwn only to close it after attempting to add it to the subscription manager
```

Comment:
Thanks for adding the context info here: to make sure I understand correctly: is the reason why it would be closed after attempting to add it to the subscription manager because subscription manager would throw an error if you attempt to add a subscription of an existing ID?

Reply:
@thehenrytsai yes. Should I modify the comment to make that clearer?
It's really defensive programming to avoid resource allocation as mentioned.

Reply:
Done